src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((obj) => {
+    if (obj.type === 'addProperties') {
+      Object.assign(state, obj.extraData);
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const key of obj.keysToRemove) {
+        delete state[key];
+      }
+    }
+
+    if (obj.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
```

Comment:
I think it is better using here [switch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch)
don't forget about 4th point of [chacklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) default should have an error! :)